### DISCUSSION TITLE
host-zigbeed instead of multiprotocol plugin for HA

### DISCUSSION THREAD

**Discussion by dumpfheimer** [zigpy/zigpy#1460] [ZHA development]
I have a quite large ZigBee network (~140 devices) and have been fighting with connectivity issues for quite some time. The most stable situation I have gotten until now was using SkyConnect with the multiprotocol plugin from HA. I never used the Matter part but I suspect that the computer simply has a better time keeping track of the network because it is not memory constrained.

I recently tried flashing SkyConnect with a regular firmware again but stability was quickly an issue again. Lamps that have not been used recently took multiple seconds to answer.

I saw that the multiprotocol support will not be developed further which made me fear that I will not be getting any updates soon which made me "develop" a docker container that runs zigbeed and cpcd without matter and uses SkyConnect as an RPC.

It has been working fine since a few days or weeks.

My questions are:
- Does it even make sense, that the zigbeed works better than the normal firmware? From what I know the routing table size should be 200 anyway which should be enough for my 130 devices.
- Could this be something that is interesting for the public / HA or would this be an edge use case, that is too extravagant to be considered useful?
- What were the reasons HA shut down the multi protocol plugin? Would those same reasons apply to the plugin I am proposing?

I do see having large buffers and fast communication between HA and zigbeed being an advantage but to be fair I do not even know for sure that the SkyConnect is memory constrained.

**Comment by puddly** [comment]
We found that multiprotocol firmware was very buggy for the people that had issues with it and that the vendor wasn't actively fixing bugs: LQI and RSSI never worked, Green Power caused constant crashes, we had to add a custom watchdog component to the firmware to deal with crashes, and so on. Thread often requires different channels from Zigbee and channel switching significantly degrades the radio performance. In short, it's a nice concept but it's not practical and has unfixable limitations.

LQI and RSSI not working effectively disables CCA and likely significantly alters routing on the mesh. I'm not aware of any memory constraints on the SkyConnect, especially with only 140 devices (only routing neighbors and end devices that are children of the coordinator are kept in-memory, everything else is basically ignored).

I would figure out why you're having issues with devices. Maybe run <https://github.com/zigpy/zigpy-cli/pull/49> and post your results?

**Comment by Hedda** [comment]
@dumpfheimer recommend also make sure follow all the generic best practice tips regarding avoiding EMI interference and adding many ”known good” Zigbee Router devices spread out in strategic before troubleshooting any deeper:

https://www.home-assistant.io/integrations/zha#zigbee-interference-avoidance-and-network-rangecoverage-optimization

https://www.home-assistant.io/integrations/zha#best-practices-to-avoid-pairingconnection-difficulties

https://www.home-assistant.io/integrations/zha#using-router-devices-to-add-more-devices

https://www.home-assistant.io/integrations/zha#defining-zigbee-channel-to-use

**Comment by dumpfheimer** [comment]
Thank you for your responses and sorry for the delay in mine.

I do not think EMI is the issue. It should affect the network regardless of where the zigbee stack is running - correct me if I'm wrong.

@puddly are you sure the stick is not memory constrained?
I have now flased the latest firmware from https://github.com/NabuCasa/silabs-firmware-builder/releases/tag/v2024.8.20 and reset everything to default and I get this error at startup:

```
DEBUG (MainThread) [bellows.ezsp] Setting config CONFIG_MULTICAST_TABLE_SIZE = 16 (old value 8)
DEBUG (MainThread) [bellows.ezsp.protocol] Sending command  setConfigurationValue: () {'configId': <EzspConfigId.CONFIG_MULTICAST_TABLE_SIZE: 6>, 'value': 16}
DEBUG (MainThread) [bellows.ezsp.protocol] Received command setConfigurationValue: {'status': <EzspStatus.ERROR_OUT_OF_MEMORY: 53>}
DEBUG (MainThread) [bellows.ezsp] Could not set config EzspConfigId.CONFIG_MULTICAST_TABLE_SIZE = 16: EzspStatus.ERROR_OUT_OF_MEMORY
```
or
```
DEBUG (MainThread) [bellows.ezsp] Setting config CONFIG_ADDRESS_TABLE_SIZE = 16 (old value 8)
DEBUG (MainThread) [bellows.ezsp.protocol] Sending command  setConfigurationValue: () {'configId': <EzspConfigId.CONFIG_ADDRESS_TABLE_SIZE: 5>, 'value': 16}
DEBUG (MainThread) [bellows.ezsp.protocol] Received command setConfigurationValue: {'status': <EzspStatus.ERROR_OUT_OF_MEMORY: 53>}
DEBUG (MainThread) [bellows.ezsp] Could not set config EzspConfigId.CONFIG_ADDRESS_TABLE_SIZE = 16: EzspStatus.ERROR_OUT_OF_MEMORY

```

Although in general the host-stack-variant seems to be a bit slower, I do not have the issue of completely unresponsive devices.
After switching to the NCP firmware (and waiting for the network to stabalize) I am getting a bunch of ZIGBEE_DELIVERY_FAILED.
